hathor/p2p/manager.py

```diff
@@ -661,20 +665,43 @@ def listen(self, description: str, use_ssl: Optional[bool] = None) -> IStreamSer
 
         factory = NetfilterFactory(self, factory)
 
-        self.log.info('listen on', endpoint=description)
-        endpoint.listen(factory)
+        self.log.info('trying to listen on', endpoint=description)
+        deferred: Deferred[IListeningPort] = endpoint.listen(factory)
+        deferred.addCallback(self._on_listen_success, description)
+
+    def _on_listen_success(self, listening_port: IListeningPort, description: str) -> None:
+        """Callback to be called when listening to an endpoint succeeds."""
+        self.log.info('success listening on', endpoint=description)
+        address = listening_port.getHost()
+
+        if not isinstance(address, (IPv4Address, IPv6Address)):
+            self.log.error(f'unhandled address type for endpoint "{description}": {str(type(address))}')
+            return
```

Comment:
Before, we were using `getattr()` instead of correctly handling the types. Should we handle anything other than Twisted's `IPv4Address` and `IPv6Address`?

Reply:
I think `IPv4Address` and `IPv6Address` should be the only 2 we support (even though IPv6 support is not complete).